### ISSUE TITLE
@available for iOS12, but uses types from iOS13.

### ISSUE THREAD

**Issue by drekka** [swift-server/swift-service-lifecycle#122]
I'm trying to use the Hummingbird server as a test server for UI testing. After adding Hummingbird and trying to compile I'm getting a series of compilation errors from the swift-service-lifecycle code. One of which looks like this:

```
checkouts/swift-service-lifecycle/Sources/Lifecycle/Lifecycle.swift:103:13: error: 'Task' is only available in iOS 13.0 or newer
            Task {
            ^
```

Looking at `Lifecycle.swift` I can see this code:

```swift
#if compiler(>=5.5) && canImport(_Concurrency)
@available(macOS 12.0, *)
extension LifecycleHandler {
    public init(_ handler: @escaping () async throws -> Void) {
        self = LifecycleHandler { callback in
            Task {
                do {
                    try await handler()

```

Now if I'm reading this right, it's saying that it should be available for iOS12 (which is what this project is still supporting :-( ) but it's using `Task` which only came with iOS13 as the error indicates.

How do I fix this?

**Comment by ktoso** [comment]
That would be a bug in the availability annotation here actually - we should fix that.

We should use `@available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)` I can send in a PR about this.

**Comment by ktoso** [comment]
Huh wait, these are actually already correct on main: https://github.com/swift-server/swift-service-lifecycle/blob/main/Sources/Lifecycle/Lifecycle.swift#L173

```
#if canImport(_Concurrency) && compiler(>=5.5.2)
@available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
extension LifecycleStartHandler {
```
